### PR TITLE
Security refactor

### PR DESCRIPTION
**Pull request by orenbm**
#### What does this PR do?

- Refactors Security class to a ValidateSecurity command class
- Removed redundant `get_login_by_id` class

#### What ticket does this PR close?
Connected to #966 
Connected to #960

### CODE REVIEW

File: app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb [hunk 1]
```diff
@@ -79,8 +79,12 @@ def conjur_username
...
 
         def validate_security
-          @validate_security.(input: @authenticator_input,
-            enabled_authenticators: @enabled_authenticators)
+          @validate_security.(
+            webservice: @authenticator_input.webservice,
+              account: @authenticator_input.account,
+              user_id: @authenticator_input.username,
+              enabled_authenticators: @enabled_authenticators
+          )
```
Comment (jonahx) [inline app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb]: This whole class is looking nice.  Was trivial for me to follow the logic.

File: app/domain/authentication/validate_security.rb [hunk 2]
```diff
@@ -1,15 +1,77 @@
...
+    end
+
+    private
+
+    def default_conjur_authn?
+      @webservice.authenticator_name ==
+        ::Authentication::Common.default_authenticator_name
+    end
+
+    def validate_webservice_exist
```
Comment (jonahx) [inline app/domain/authentication/validate_security.rb]: english grammar nit.  should be: `validate_webservice_exists`

File: spec/app/domain/authentication/validate_security_spec.rb [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+          user_id: 'some-user',
+          enabled_authenticators: blank_env
+      )
+    end
+
+    it "the default Conjur authenticator is included in whitelisted webservices" do
+      expect { subject }.to_not raise_error
+    end
+  end
+end
```
Comment (jonahx) [inline spec/app/domain/authentication/validate_security_spec.rb]: Excellent work on this spec.  This is a _textbook_ demonstration of how good design improves UTs.

File: app/domain/authentication/validate_security.rb [hunk 4]
```diff
@@ -1,15 +1,77 @@
...
+      validate_webservice_exist
+      validate_user_is_defined
+      validate_user_has_access
+    end
+
+    private
+
+    def default_conjur_authn?
+      @webservice.authenticator_name ==
+        ::Authentication::Common.default_authenticator_name
```
Comment (jtuttle) [inline app/domain/authentication/validate_security.rb]: I believe this should be a dependency as well. Is that correct @jonahx?

File: app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb [hunk 5]
```diff
@@ -79,8 +79,12 @@ def conjur_username
         end
```
[the remark this answers is not in the document]
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb]: nice break, thanks!

File: app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb]: yes i tried to fix that but in my IDE (RubyMine) this is how the indent multilines. I also don't like it but can't really fix it

File: app/domain/authentication/validate_security.rb (hunk 2 above)
Comment (orenbm) [reply app/domain/authentication/validate_security.rb]: correct :)

File: spec/app/domain/authentication/validate_security_spec.rb (hunk 3 above)
Comment (orenbm) [reply spec/app/domain/authentication/validate_security_spec.rb]: Thanks!

File: app/domain/authentication/validate_security.rb [hunk 6]
```diff
@@ -1,15 +1,77 @@
...
+
+    def validate_user_is_defined
+      raise NotDefinedInConjur, @user_id unless user_role
+    end
+
+    def validate_user_has_access
+      # Ensure user has access to the service
+      has_access = user_role.allowed_to?('authenticate', webservice_resource)
+      unless has_access
+        Rails.logger.debug("[OIDC] User '#{@user_id}' is not authorized to " \
```
[the remark this answers is not in the document]
Comment (orenbm) [reply app/domain/authentication/validate_security.rb]: nice catch

File: app/domain/authentication/validate_security.rb (hunk 4 above)
Comment (orenbm) [reply app/domain/authentication/validate_security.rb]: I actually don't think so. it's referring to a static string in another part of the module, not something that should be mocked.

Comment (orenbm) [reply app/domain/authentication/validate_security.rb]: @jonahx @jtuttle

**Pull request merged**